http/cert/src/main/java/org/wildfly/security/http/cert/ClientCertAuthenticationMechanism.java

```diff
@@ -141,10 +141,9 @@ private boolean attemptAuthentication(HttpServerRequest request, Function<Securi
                 authorizedFunction = cacheCallback::isAuthorized;
                 authorizeCallBack = cacheCallback;
             } else {
-                String name = evidence.getDecodedPrincipal().getName();
-                AuthorizeCallback plainCallback = new AuthorizeCallback(name, name);
-                authorizedFunction = plainCallback::isAuthorized;
-                authorizeCallBack = plainCallback;
+                PrincipalAuthorizeCallback principalCallback = new PrincipalAuthorizeCallback(evidence.getDefaultPrincipal());
```

Comment:
Should use `evidence.getDecodedPrincipal()` instead (`getDecodedPrincipal()` also takes into account any evidence decoders that have been configured).

Reply:
Done!